tests/round_labeling/test_all_paths_comprehensive.py

```
@@ -0,0 +1,439 @@
+        paths = []
+        path_gen = generate_all_paths(TRANSACTION_GRAPH, self.constraints)
+        
+        # More efficient: collect all paths up to start_idx + batch_size
+        # then slice the result
+        collected = 0
+        for i, path in enumerate(path_gen):
+            if i >= start_idx:
+                paths.append(path)
+                if len(paths) >= batch_size:
+                    break
+            collected += 1
```

Ruff (0.11.9)

Use `enumerate()` for index variable `collected` in `for` loop

(SIM113)